src/rust/engine/process_executor/src/main.rs

```diff
@@ -160,6 +161,13 @@ fn main() {
             .required(false)
             .help("Symlink a JDK from .jdk in the working directory. For local execution, symlinks to the value of this flag. For remote execution, just requests that some JDK is symlinked if this flag has any value. https://github.com/pantsbuild/pants/issues/6416 will make this less weird in the future.")
       )
+      .arg(
+        Arg::with_name("target-platform")
+            .long("target-platform")
+            .takes_value(true)
+            .required(true)
+            .help("The name of the platform that this request's output is compatible with. Options are 'linux', 'darwin', or 'none' (which indicates either)")
```

Comment:
Rather than actually configuring which platform something runs on, this instead sortof "asserts" that the command is compatible with a particular platform, right? Might be good to clarify the relationship between flags maybe.

Reply:
Yeah. That is true. I suppose this tool should also be updated to use the speculating command runner. Then this flag could do more than just annotate which platform the request is meant for. We could also get away with just hardcoding this to Platform::None if we don't want to update the tool.